tests/unit/synapseclient/extensions/unit_test_create_json_schema.py

```diff
@@ -0,0 +1,1341 @@
+            validator.validate({"name": value})
+
+
+@pytest.mark.parametrize(
+    "node_name, expected_schema, valid_values, invalid_values",
+    [
+        # If is_required is True, no type is added
+        (
+            "Enum",
+            {"oneOf": [{"enum": ["ab", "cd", "ef", "gh"], "title": "enum"}]},
+            ["ab"],
+            [1, "x", None],
+        ),
+        # If is_required is False, "null" is added as a type
+        (
+            "EnumNotRequired",
+            {
+                "oneOf": [
+                    {"enum": ["ab", "cd", "ef", "gh"], "title": "enum"},
+                    {"type": "null", "title": "null"},
+                ],
+            },
+            ["ab", None],
+            [1, "x"],
+        ),
+    ],
+    ids=["Required", "Not required"],
+)
+def test_create_enum_property(
+    node_name: str,
+    expected_schema: dict[str, Any],
+    valid_values: list[Any],
+    invalid_values: list[Any],
+    test_nodes: dict[str, Node2],
+) -> None:
+    """Test for _create_enum_property"""
+    schema = _create_enum_property(test_nodes[node_name])
+    assert schema == expected_schema
+    full_schema = {"type": "object", "properties": {"name": schema}, "required": []}
+    validator = Draft7Validator(full_schema)
+    for value in valid_values:
+        validator.validate({"name": value})
+    for value in invalid_values:
+        with pytest.raises(ValidationError):
+            validator.validate({"name": value})
+
+
+@pytest.mark.parametrize(
+    "node_name, expected_schema, valid_values, invalid_values",
+    [
+        ("NoRulesNotRequired", {}, [None, 1, ""], []),
+        # If property_type is given, it is added to the schema
+        (
+            "String",
+            {"type": "string"},
+            [""],
+            [1, None],
+        ),
+        # If property_type is given, and is_required is False,
+        # type is set to given property_type and "null"
+        (
+            "StringNotRequired",
+            {
+                "oneOf": [
+                    {"type": "string", "title": "string"},
+                    {"type": "null", "title": "null"},
+                ],
+            },
+            [None, "x"],
+            [1],
+        ),
+        # If is_required is True '"not": {"type":"null"}' is added to schema if
+        # property_type is not given
+        (
+            "NoRules",
+            {"not": {"type": "null"}},
+            ["x", 1],
+            [None],
+        ),
+        (
+            "InRange",
+            {
+                "type": "number",
+                "minimum": 50,
+                "maximum": 100,
+            },
+            [50, 75, 100],
+            [None, 0, 49, 101],
+        ),
+    ],
+    ids=[
+        "Not required, no type",
+        "Required, string type",
+        "Not required, string type",
+        "Required, no type",
+        "Required, number type",
+    ],
+)
+def test_create_simple_property(
```

Comment:
Similarly to above, I updated this test to use `test_nodes_column_types` so that the generated schema can be used for validation.